docs/source/user/workspaces.rst

```diff
@@ -0,0 +1,115 @@
+- `save`, `save as`, `import`, and `export` can load and store the workspace to/from the file system (contained within the Jupyter root directory); `save` will save the workspace to the most recently saved file.
+
+In the sidebar the current workspace is indicated with check mark (✓). A different workspace can be opened by clicking on the corresponding sidebar item. Opening context menu (right click) over the workspace item in the sidebar will present actions available for management of that workspace:
+
+.. image:: ../images/workspaces-sidebar.png
+   :align: center
+   :class: jp-screenshot
+   :alt: The context menu opened over workspaces sidebar
+
+.. _workspaces-cli:
+
+Managing Workspaces (CLI)
+-------------------------
+
+JupyterLab provides a command-line interface for workspace ``import`` and
+``export``:
+
+.. code-block:: bash
+
+  $ # Exports the default JupyterLab workspace
+  $ jupyter lab workspaces export
+  {"data": {}, "metadata": {"id": "/lab"}}
+  $
+  $ # Exports the workspaces named `foo`
+  $ jupyter lab workspaces export foo
+  {"data": {}, "metadata": {"id": "/lab/workspaces/foo"}}
+  $
+  $ # Exports the workspace named `foo` into a file called `file_name.json`
+  $ jupyter lab workspaces export foo > file_name.json
+  $
+  $ # Imports the workspace file `file_name.json`.
+  $ jupyter lab workspaces import file_name.json
+  Saved workspace: <workspaces-directory>/labworkspacesfoo-54d5.jupyterlab-workspace
+
+The ``export`` functionality is as friendly as possible: if a workspace does not
+exist, it will still generate an empty workspace for export.
+
+The ``import`` functionality validates the structure of the workspace file and
+validates the ``id`` field in the workspace ``metadata`` to make sure its URL is
+compatible with either the ``workspaces_url`` configuration or the ``page_url``
+configuration to verify that it is a correctly named workspace or it is the
+default workspace.
+
+
+Workspace File Format
+---------------------
+
+A workspace file in a JSON file with a specific spec.
+
+
+There are two top level keys requires, `data`, and `metadata`.
+
+The `metadata` must be a mapping with an `id`
+key that has the same value as the ID of the workspace. This should also be the relative URL path to access the workspace,
+like `/lab/workspaces/foo`. Additionally, `metadata` may contain `created` and `last_modified` fields with date and time creation and most recent modification, respectively. The date and time are encoded using ISO 8601 format.
+
+The `data` key maps to the initial state of the ``IStateDB``. Many plugins look in the State DB for the configuration.
+Also any plugins that register with the ``ILayoutRestorer`` will look up all keys in the State DB
+that start with the `namespace` of their tracker before the first ``:``. The values of these keys should have a `data`
+attribute that maps.
```

Comment:
This is not clear. We should use "state database" as the shorthand for `IStateDB` (even though it is longer in characters, it is easier to read). Although this refers to "many plugins", it's not clear what a user would find important about this. The final sentence also trails off; it is not clear what is mapped to what.